server/middleware.go

```diff
@@ -102,3 +104,60 @@ func withLogging(
 		log.Info("request", args...)
 	}
 }
+
+// withCORS is a middleware that adds CORS headers to responses.
+// It intercepts OPTIONS requests for handling CORS preflight requests.
```

Comment:
why is this needed? Is it because our POST returns binary data?
<img width="704" alt="image" src="https://github.com/user-attachments/assets/d52097e1-bc1a-43cb-8c79-6961c64855a2" />

Reply:
i think this is one of those things we should just handle now so that if you add more endpoints in the future, with other content types it's already adjusted.